src/main/java/de/yuna/berlin/nativeapp/services/http/model/HttpRequest.java

```diff
@@ -0,0 +1,322 @@
+    }
+
+    public boolean containsQueryParam(final String key) {
+        return queryParams.containsKey(key);
+    }
+
+    public String getQueryParam(final String key) {
+        Object value = queryParams.get(key);
+        return value != null ? value.toString() : null;
+    }
+
+    public boolean exactMatch(final String path) {
+        return this.path.startsWith(path);
+    }
+
+    public boolean match(final String pathToMatch) {
+
+        String[] partsToMatch = pathToMatch.split("/");
+        String[] parts = path.split("/");
+
+        if (exactMatch(pathToMatch) && pathToMatch.contains("{")) return true;
+
+        if (partsToMatch.length != parts.length) return false;
+
+        if (pathParams == null)
+            pathParams = new TypeMap();
+        else
+            pathParams.clear();
+        for (int i = 0; i < partsToMatch.length; i++) {
+            if (!partsToMatch[i].equals(parts[i])) {
+                if (partsToMatch[i].startsWith("{")) {
+                    String key = partsToMatch[i].substring(1, partsToMatch[i].length() - 1);
+                    pathParams.put(key, parts[i]);
+                } else {
+                    return false;
+                }
+            }
+        }
+
+        return true;
+    }
+
+
+    public TypeMap getPathParams() {
+        return pathParams;
+    }
+
+    public String getPathParam(final String key) {
+        return pathParams.get(key).toString();
+    }
+
+    public String getHeader(final String name) {
+        String value = headers.get(name).toString();
+        if (value.startsWith("[")) {
+            value = value.substring(1, value.length() - 1);
+        }
+        return value;
+    }
+
+    public TypeMap getHeaders() {
+        return headers;
+    }
+
+    public boolean containsHeader(final String name) {
+        return headers.containsKey(name.toLowerCase());
+    }
+
+    public String host() {
+        return exchange.getLocalAddress().getHostName();
+    }
+
+    public int port() {
+        return exchange.getLocalAddress().getPort();
+    }
+
+    public String protocol() {
+        return exchange.getProtocol();
+    }
+
+    public String userAgent() {
+        return getHeader(HttpHeaders.USER_AGENT);
+    }
+
+    public String getAuthorisationToken() {
+        String value = getHeader(HttpHeaders.AUTHORIZATION);
+        if (value.startsWith("Bearer ")) {
+            value = value.substring("Bearer ".length());
+        }
+        return value;
+    }
+
+    public List<String> getPreferredLanguages() {
+        String acceptLanguageHeader = getHeader(HttpHeaders.ACCEPT_LANGUAGE);
+        if (acceptLanguageHeader == null || acceptLanguageHeader.isEmpty()) {
+            return Collections.emptyList();
+        }
+
+        if (acceptLanguageHeader.startsWith("[")) {
+            acceptLanguageHeader = acceptLanguageHeader.substring(1, acceptLanguageHeader.length() - 1);
+        }
+
+        Set<String> displayLanguages = new HashSet<>();
```

Comment:
It seems you are not taking the order in account. e.g. `"en-US,en;q=0.8,fr-CA;q=0.9,fr;q=0.7"`

Maybe this helps:
```
Arrays.stream("en-US,en;q=0.8,fr-CA;q=0.9,fr;q=0.7".split(","))
            .map(s -> s.split(";q="))
            .sorted(Comparator.comparing(parts -> parts.length > 1 ? Double.parseDouble(parts[1]) : 1.0, Comparator.reverseOrder()))
            .map(parts -> Locale.forLanguageTag(parts[0]))
            .forEach(System.out::println);
```

Reply:
Current output
PreferredLanguage: [English, German]

Should we print above or 
[en_CA, en, en_US, de_DE, de]

Reply:
you need to verify my example, but in this example the user gets an ordered list of Locales which have the methods for display name, iso name, country etc. already. e.g. `.forEach(locale -> System.out.println("Language [" + locale.getLanguage() + "] Country [" + locale.getCountry() + "]"));` will print

```
Language [en] Country [US]
Language [fr] Country [CA]
Language [en] Country []
Language [fr] Country []
```